### ISSUE TITLE
Actual training for target dataset after view predictor has been trained.

### ISSUE THREAD

**Issue by ri-sal** [pse-ecn/pose-sensitive-embedding#14]
Hi,

I have trained the Views Predictor using the command mentioned in the documentation with the RAP dataset:

' python3 trainer_views.py --output=<output directory> --data=<dataset directory> --dataset-name=rap --batch-size=16 --num-epochs=100 --network-name=resnet_v1_50_views --initial-checkpoint=<path to imagenet checkpoint or another checkpoint you want to load> --checkpoint-exclude-scopes=resnet_v1_50/logits --trainable-scopes=resnet_v1_50/3Views'

How can we transfer this result to train the target dataset (market1501) finally? What would be the checkpoint/scopes for this stage?

(I am interested in training and evaluating the views only method in this case)

Thank you.

**Comment by ssarfraz** [comment]
python3 trainer_preid.py --output=<output directory> --data=<dataset directory> --dataset-name=market1501 --batch-size=16 --num-epochs=100 --network-name=resnet_v1_50_views --initial-checkpoint=<path to model with RAP pre-trained views predictor> --checkpoint-exclude-scopes=resnet_v1_50/logits --trainable-scopes=resnet_v1_50/logits,resnet_v1_50/pre_logits,resnet_v1_50/3ViewBranches --no-evaluation

**Comment by ssarfraz** [comment]
here initial-checkpoint is your model with trained Views predictor

**Comment by ri-sal** [comment]
Hi, I have been trying to reproduce the results of your Views Only approach but I am not able to get the same results. These were my steps for training:

**Training a Resnet-50 Baseline on Market1501**
python3 trainer_preid.py --output='output_prid' --data='/mnt/Datasets/Incoming/Surveillance/PRID/Market-1501/data/Market-1501-v15.09.15/' --dataset-name=market1501 --batch-size=16 --num-epochs=100 --network-name=resnet_v1_50 --initial-checkpoint='./checkpoints/resnet_v1_50.ckpt' --checkpoint-exclude-scopes=resnet_v1_50/logits --trainable-scopes=resnet_v1_50/logits --no-evaluation

**Training the Views Predictor with RAP**

(here i gave the checkpoint from the previous step and I had to use --ignore missing variables as suggested in issue #12 )

python3 trainer_views.py --output='trial2_output_prid_views' --data='/mnt/Datasets/Incoming/Surveillance/PRID/RAP/RAP_dataset' --dataset-name=rap --batch-size=16 --num-epochs=100 --network-name=resnet_v1_50_views --initial-checkpoint='/home/srl4hi/PycharmProjects/pose-sensitive-embedding-master_1.3/output_prid/model.ckpt-86534' --checkpoint-exclude-scopes=resnet_v1_50/logits --trainable-scopes=resnet_v1_50/3Views --ignore-missing-variables

**Training the Views Predictor on Market**

(here i gave the checkpoint from the previous step)

python3 trainer_preid.py --output='trial2_output_prid_views_2' --data='/mnt/Datasets/Incoming/Surveillance/PRID/Market-1501/data/Market-1501-v15.09.15/' --dataset-name=market1501 --batch-size=16 --num-epochs=100 --network-name=resnet_v1_50_views --initial-checkpoint='/home/srl4hi/PycharmProjects/pose-sensitive-embedding-master_1.3/trial2_output_prid_views/model.ckpt-207900' --checkpoint-exclude-scopes=resnet_v1_50/logits --trainable-scopes=resnet_v1_50/logits,resnet_v1_50/pre_logits,resnet_v1_50/3ViewBranches --no-evaluation

**Prediction**
python3 predictor_preid.py --model-dir='trial2_output_prid_views_2' --data='/mnt/Datasets/Incoming/Surveillance/PRID/Market-1501/data/Market-1501-v15.09.15/' --dataset-name=market1501 --batch-size=128 --network-name=resnet_v1_50_views

**This is the result that I got.**
MatlabEvaluationResult(mAP=31.119823695326907, rank1=58.10570071258907, rank5=80.76009501187649, rank10=87.44061757719716, rank50=96.16983372921615)


Please let me know if I am missing a step in the training or using the wrong initial checkpoints. 

Thank you for your time.

**Comment by ssarfraz** [comment]
@andreas-eberle , Andreas, can you take a look at his steps and see what is he missing

**Comment by huangzhenjie** [comment]
you only fine-tune the network. you could try to use command "python3 trainer_preid.py --output='trial2_output_prid_views_2' --data='/mnt/Datasets/Incoming/Surveillance/PRID/Market-1501/data/Market-1501-v15.09.15/' --dataset-name=market1501 --batch-size=16 --num-epochs=150 --network-name=resnet_v1_50_views --initial-checkpoint=(checkpoint from the  step[Training the Views Predictor on Market])  --no-evaluation"
